arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -18,140 +18,228 @@ import ArcGIS
-        // NOTE: Never hardcode login information in a production application. This is done solely for the sake of the sample.
+        UIApplication.shared.showProgressHUD(message: "Loading service geodatabase…")
+        // NOTE: Never hardcode login information in a production application.
+        // This is done solely for the sake of the sample.
         serviceGeodatabase.credential = AGSCredential(user: "viewer01", password: "I68VGU^nMurF")
         serviceGeodatabase.load { [weak self] error in
             guard let self = self else { return }
-            // The  gas device layer ./0 and gas line layer ./3 are created from the service geodatabase.
+            // The gas device layer ./0 and gas line layer ./3 are created
+            // from the service geodatabase.
             if let gasDeviceLayerTable = self.serviceGeodatabase.table(withLayerID: 0),
                let gasLineLayerTable = self.serviceGeodatabase.table(withLayerID: 3) {
-                self.layers = [gasLineLayerTable, gasDeviceLayerTable].map(AGSFeatureLayer.init)
+                let layers = [gasLineLayerTable, gasDeviceLayerTable].map(AGSFeatureLayer.init)
                 // Add the utility network feature layers to the map for display.
-                self.mapView.map?.operationalLayers.addObjects(from: self.layers)
+                self.mapView.map?.operationalLayers.addObjects(from: layers)
+                self.layers = layers
                 self.loadUtilityNetwork()
             } else if let error = error {
+                UIApplication.shared.hideProgressHUD()
                 self.presentAlert(error: error)
+            } else {
+                UIApplication.shared.hideProgressHUD()
             }
         }
     }
     
     /// Load the utility network.
     func loadUtilityNetwork() {
-        setStatus(message: "Loading utility network…")
+        UIApplication.shared.showProgressHUD(message: "Loading utility network…")
         // Load the utility network to be ready to run a trace against it.
         utilityNetwork.load { [weak self] error in
             guard let self = self else { return }
-            if let error = error {
-                self.setStatus(message: "Loading utility network failed.")
+            let errorMessage = "Failed to load utility network."
+            guard error == nil else {
+                UIApplication.shared.hideProgressHUD()
+                self.presentAlert(error: error!)
+                self.setStatus(message: errorMessage)
+                return
+            }
+            // Create a default starting location.
+            guard let startingLocation = self.makeStartingLocation() else {
+                UIApplication.shared.hideProgressHUD()
+                self.presentAlert(message: "Failed to create starting location.")
+                self.setStatus(message: errorMessage)
+                return
+            }
+            self.utilityNetworkDidLoad(startingLocation: startingLocation)
```

Comment:
The spirit behind `guard` is that it checks for a condition that must be met in order for forward progress to be made (hence the `else` clause must exit the current scope). When work is done in an `else` clause, forward progress is being made either way, making the guard statement less intuitive and thereby making the code harder to reason about, not easier. In almost all cases, the `else` clause of `guard` statement should be a single line. If it isn't, that's a pretty good indication to use an `if` statement instead.

```suggestion
            if let error = error {
                UIApplication.shared.hideProgressHUD()
                self.presentAlert(error: error)
                self.setStatus(message: errorMessage)
            } else if let startingLocation = self.makeStartingLocation() {
                self.utilityNetworkDidLoad(startingLocation: startingLocation)
            } else {
                UIApplication.shared.hideProgressHUD()
                self.presentAlert(message: "Failed to create starting location.")
                self.setStatus(message: errorMessage)
            }
```